src/server.rs

```diff
@@ -219,11 +219,19 @@ fn recv_loop(peer_id: usize, stream: &TcpStream, server_tx: Sender<Event>) -> Re
     let msg = Message::New(stream.try_clone()?);
     server_tx.send(Event { peer_id, msg })?;
 
+    let mut first_line = true;
     for line in BufReader::new(stream).lines() {
+        if let Err(ref e) = line {
+            warn!("{}: recv error: {:?} - {}", peer_id, e.kind(), e);
+            if first_line && e.kind() == std::io::ErrorKind::InvalidData {
+                warn!("InvalidData on first line may indicate client attempted to connect using SSL when server expects unencrypted communication.")
+            }
+        }
         let line = line.with_context(|| format!("{}: recv failed", peer_id))?;
```

Comment:
Was thinking of something like this, not sure if it's better:

```rust
let line = line.map_err(|error| {
    // warn!() if it's not redundant
    if first_line && error.kind() == std::io::ErrorKind::InvalidData {
        warn!("InvalidData on first line may indicate client attempted to connect using SSL when server expects unencrypted communication.")
    }
    first_line = false;
    error.context(format!("{}: recv failed", peer_id))
})?
```

Reply:
I tried doing that but `error.context()` doesn't exist, would need the `Result`.

Reply:
I avoided the extra `warn!` unless it's not an `InvalidData` error in the newest version.

Reply:
Ah, didn't realize `error` is not `anyhow::Error`.